_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -102,4 +102,154 @@ test.describe('Snapshot Repositories', () => {
+      await page.getByRole('button', { name: 'Select architecture' }).click();
+      await page.getByRole('option', { name: 'aarch64' }).click();
+      await page.getByRole('button', { name: 'Select version' }).click();
+      await page.getByRole('option', { name: 'el9' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await row.click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByTestId('use-latest-snapshot-radio').click();
+      await page.getByRole('radio', { name: 'Use latest content' }).check();
+      await page.getByRole('button', { name: 'Next' }).click();
+      await page.getByPlaceholder('Enter name').fill(`${templateName}`);
+      await page.getByPlaceholder('Description').fill('Template test');
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Create other options' }).click();
+      await page.getByText('Create template only', { exact: true }).click();
+      const templateRow = await getRowByNameOrUrl(page, templateName);
+      await expect(templateRow.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      // Verify the template is created and uses the latest snapshot
+      await expect(templateRow.getByText('Use latest')).toBeVisible();
+    });
+
+    // Test deletion of a single snapshot.
+    await test.step('Delete a single snapshot', async () => {
+      await navigateToRepositories(page);
+      const row = await getRowByNameOrUrl(page, repoName);
+      await row.getByLabel('Kebab toggle').click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      await expect(page.getByLabel('SnapshotsView list of').locator('tbody')).toBeVisible();
+      await page
+        .getByTestId('snapshot_list_table')
+        .locator('tbody tr')
+        .first()
+        .getByLabel('Kebab toggle')
+        .click();
+      await page.getByRole('menuitem', { name: 'Delete' }).click();
+      await expect(page.getByText('Remove snapshots?')).toBeVisible();
+      await page.getByText('Remove', { exact: true }).click();
+      const snapshotPrefix = new Date()
+        .toLocaleString('en-GB', {
+          day: '2-digit',
+          month: 'short',
+          year: 'numeric',
+          hour12: false,
+        })
+        .replace(',', ''); // Format: 20 May 2025
+      const targetRows = page.getByRole('row').filter({ has: page.getByText(snapshotPrefix) });
+      await expect(targetRows).toHaveCount(3);
+      await page.getByText('Close').click();
+    });
+
+    await test.step('Bulk delete snapshot', async () => {
+      // Test bulk deletion of multiple snapshots.
+      // Before bulk delete we need to wait for previous deletion to finish or it will fail.
+      await page.waitForTimeout(6000);
```

Comment:
We need to add a timeout here, as the bulk delete operation fails by mistakenly recognizing it as the same delete action from the previous single delete test step.